pkg/patch/patch.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package patch
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/trustbloc/sidetree-core-go/pkg/docutil"
+)
+
+// Action defines action of document patch
+type Action string
+
+const (
+	// Replace captures enum value "replace"
+	Replace Action = "replace"
+
+	// JSONPatch captures enum value "json-patch"
+	JSONPatch Action = "ietf-json-patch"
+)
+
+// Key defines key that will be used to get document patch information
+type Key string
+
+const (
+
+	// DocumentKey captures  "document" key
+	DocumentKey Key = "document"
+
+	// PatchesKey captures "patches" key
+	PatchesKey Key = "patches"
+
+	// ActionKey captures "action" key
+	ActionKey Key = "action"
+)
+
+// Patch defines generic patch structure
+type Patch map[Key]interface{}
+
+// NewReplacePatch creates new replace patch
+func NewReplacePatch(document string) Patch {
+	patch := make(Patch)
+	patch[ActionKey] = Replace
+	patch[DocumentKey] = document
+
+	return patch
+}
+
+// NewJSONPatch creates new generic update patch (will be used for generic updates)
+func NewJSONPatch(patches string) Patch {
+	patch := make(Patch)
+	patch[ActionKey] = JSONPatch
+	patch[PatchesKey] = patches
+
+	return patch
+}
+
+// GetStringValue returns string value for specified key or "" if not found or wrong type
+func (p *Patch) GetStringValue(key Key) string {
```

Comment:
You can do (p Patch) instead of (p *Patch) since patch is a map and map is already a pointer.

Reply:
Fixed